.github/workflows/build.yml

```diff
@@ -5,35 +5,67 @@ on: [push, pull_request]
 jobs:
   build-macos:
     runs-on: macOS-latest
+    env:
+      CCACHE_COMPRESS: 1
+      CCACHE_TEMPDIR: /tmp/.ccache-temp
     steps:
     - uses: actions/checkout@v1
       with:
         submodules: recursive
+    - name: ccache
+      uses: actions/cache@v1
+      with:
+        path: ~/.ccache
```

Comment:
Please ELI5:
does the same path for all the *nix instances mean, that the exact files are shared? Or are these actually different files, whose path is just named the same way? I'm trying to settle whether you share the cached files across OSes, which shouldn't be done. Perhaps if this is the case, changing the paths to use individual ones would also be the cure?

Reply:
> does the same path for all the *nix instances mean, that the exact files are shared?

These are all separate VMs. No files are being shared between different OS or different jobs. Each of these jobs (build-macos, build-windows, build-ubuntu, libwallet-ubuntu) share one cache.

See: `restore-keys: ccache-windows-build-` for the Windows build job.